ecommers/lib/core/mixins/items_loading_notifier.dart

```diff
@@ -0,0 +1,31 @@
+import 'dart:async';
+
+import 'package:flutter/material.dart';
+
+import 'package:ecommers/core/app_services/index.dart';
+import 'package:ecommers/core/services/dependency_service.dart' as services;
+
+mixin ItemsLoadingNotifier on ChangeNotifier {
+  @protected
+  final Paginator paginator = services.paginator;
```

Comment:
Why do you get paginator like this?

Reply:
because it registered as a factory in get it